query-node/mappings/src/forum.ts

```diff
@@ -307,7 +309,7 @@ export async function forum_ThreadCreated(ctx: EventContext & StoreContext): Pro
       createdAt: eventTime,
       updatedAt: eventTime,
       description: bytesToString(pollInput.unwrap().description),
-      endTime: new Date(pollInput.unwrap().end_time.toNumber()),
+      endTime: new Date(toNumber(pollInput.unwrap().end_time, new Date('3000-01-01').getTime())),
```

Comment:
I don't think the query-node should be opinionated about what the max timestamp should be. I know you picked a number far in the future, but still. Might as well just make it INT32MAX.

Perhaps we should change the schema to allow storing full u64? This general issue has been covered before?

Reply:
> It should be noted that the maximum Date is not of the same value as the maximum safe integer (Number.MAX_SAFE_INTEGER is 9,007,199,254,740,991). Instead, it is defined in ECMA-262 that a maximum of ±100,000,000 (one hundred million) days relative to January 1, 1970 UTC (that is, April 20, 271821 BCE ~ September 13, 275760 CE) can be represented by the standard Date object (equivalent to ±8,640,000,000,000,000 milliseconds).
Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date

So according to this, we can cap to `8640000000000000`, but not to `MAX_SAFE_INTEGER`.
I have just chosen some very high value, since I noticed that `new Date(Number.MAX_SAFE_INTEGER)` doesn't work.

Reply:
Introduced `TIMESTAMPMAX` const in https://github.com/Joystream/joystream/pull/3299/commits/c7a129d856126767c1b2b43962de4728dddba3a2